reference/basic-architecture.md

```diff
@@ -9,61 +9,66 @@ More detailed guides to Mayastor's components, their design and internal structu
+## Component Roles
 
-### Mayastor
+ ### io-engine
+     The io-engine pods encapsulate Mayastor containers, which implement the I/O path from the block devices at the persistence layer, up to the relevant initiators on the worker nodes mounting volume claims.
+     The instance of the Mayastor running inside the container performs four major classes of functions:
+     - Presents a gRPC interface to the components which allow it to orchestrate the creation, configuration, and deletion of Mayastor-managed objects hosted by that instance.
+     - Creates and manages storage pools hosted on that node.
 
-The Mayastor pods of a deployment are its principle data plane actors, encapsulating the Mayastor containers which implement the I/O path from the block devices at the persistence layer, up to the relevant initiators on the worker nodes mounting volume claims.
+     - Creates, exports, and manages volume controller objects hosted on that node.
 
-The instance of the `mayastor` binary running inside the container performs four major classes of functions:
+     - Creates and shares replicas from storage pools hosted on that node over NVMe-TCP.
+ 
+    When an io-engine pod starts running, an init container attempts to verify connectivity to the agent-core in the namespace where Mayastor has been deployed. If a connection is established, the Mayastor container is started, and the instance performs registration with the control plane. In this way, the agent-core maintains a registry of nodes and their current state.
 
-* Present a gRPC interface to the control plane components, to allow the latter to orchestrate creation, configuration and deletion of Mayastor managed objects hosted by that instance
-* Create and manage storage pools hosted on that node
-* Create, export and manage nexus objects \(and by extension, volumes\) hosted on that node
-* Create and shares "replicas" from storage pools hosted on that node over NVMe-TCP
+    The scheduling of these pods is determined declaratively by using a DaemonSet specification. By default, a nodeSelector field is used within the pod spec to select all worker nodes to which the user has attached the label `openebs.io/engine=mayastor` as recipients of an io-engine pod. In this way, the node count and location are set appropriately to the hardware configuration of the worker nodes, and the capacity and performance demands of the cluster.
 
-When a Mayastor pod starts running, an init container attempts to verify connectivity to the NATS message bus in the Mayastor namespace. If a connection can be established the Mayastor container is started, and the Mayastor instance performs registration with MOAC over the message bus. In this way, the core agent maintains a registry of nodes \(specifically, running Mayastor instances\) and their current state.
```

Comment:
We shouldn't just remove the details on registration completely.  Moac is gone but its place is taken by the core agent.  NATS is no more but the mayastor container still registers over gRPC, and hence the control plane still maintains a registry of nodes and their state.  Also, init containers are still used _I think_, to check connectivity to etcd (needs checking)

Reply:
Okay, will check this and update the content accordingly.

Reply:
Updated the text and got it verified by Abhinandan.